src/node.cc

```diff
@@ -1510,15 +1511,16 @@ void AppendExceptionLine(Environment* env,
 
   Local<String> arrow_str = String::NewFromUtf8(env->isolate(), arrow);
 
-  if (!arrow_str.IsEmpty() && !err_obj.IsEmpty() && err_obj->IsNativeError()) {
+  if (!arrow_str.IsEmpty() && !err_obj.IsEmpty() &&
+      (!handlingFatalError || err_obj->IsNativeError())) {
```

Comment:
@cjihrig Probably not. I thought about that, but the current behaviour of printing for non-native errors is actually tested.

Printing out in that case also makes sense, because otherwise the user would receive no indicator that the output is actually an error, i.e. node could fail with just `[Object object]` as its output.

@bnoordhuis That would not have the same semantics, though. I don’t want to blow up the diff too much, but I think the actual more readable thing to do would be inverting the code paths, i.e. moving the printing inside a conditional to indicate that that is not the default route.